web/modules/custom/collection/collection.module

```diff
@@ -441,11 +441,40 @@ function collection_search_api_query_search_api_field_field_collection_content_a
 
   // Exclude custom pages from the collection overview.
   $query->addCondition('entity_bundle', 'custom_page', '<>');
+}
 
-  // Sort by sticky and then by score. The "score" field has the special name
-  // "search_api_relevance".
-  $query->sort('sticky', QueryInterface::SORT_DESC);
-  $query->sort('search_api_relevance', QueryInterface::SORT_DESC);
+/**
+ * Implements hook_search_api_solr_query_alter().
+ *
+ * Alters the Solr query for the collection content on the collection homepage
+ * by setting the sorts as follows:
+ * - Top level: Pinned content of the collection goes on top but not pinned
+ *   content of other collections shared in this collection.
+ * - 2nd level: The content score (relevancy) in the Solr index. The most
+ *   relevant are on top.
+ * - Last level: In case the score/relevancy is the same, the newest goes first.
+ */
+function collection_search_api_solr_query_alter(Query $solarium_query, QueryInterface $query) {
+  if ($query->hasTag('search_api_field_field_collection_content')) {
+    /** @var \Drupal\Core\Entity\EntityInterface $entity */
+    if ($entity = $query->getOption('search_api_field entity')) {
+      // As the 1st level cannot be stored in the Solr index, because it depends
+      // on the collection homepage context, we create a "sort by function"
+      // computed sort. The values that are used for sort are evaluated at
+      // runtime instead of being retrieved from the index. The value returned
+      // by the computed sort is TRUE if the content item is pinned and belongs
+      // to the current displayed collection and FALSE otherwise.
+      // @see https://lucene.apache.org/solr/guide/6_6/function-queries.html#function-queries
+      $computed_sort_pattern = "and(bs_sticky,query({!v='sm_entity_groups:%s'}))";
```

Comment:
Nice trick the `!v=''` to avoid adding a `qq` parameter and bypass the `can not sort on multivalued field:` error.